components/logins/src/db.rs

```diff
@@ -565,6 +565,54 @@ impl LoginDb {
         Ok(())
     }
 
+    pub fn check_valid_with_no_dupes(&self, login: &Login) -> Result<()> {
+        login.check_valid()?;
+        let _dupe_exists = self.dupe_exists(login)?;
+
+        if _dupe_exists {
+            throw!(InvalidLogin::DuplicateLogin);
+        }
+        Ok(())
+    }
+
+    pub fn dupe_exists(&self, login: &Login) -> Result<bool> {
+        Ok(self.db.query_row_named(
+            "SELECT EXISTS(
+                SELECT 1 FROM loginsL
+                WHERE is_deleted = 0
+                    AND guid <> :guid
```

Comment:
@thomcc FYI I updated the logic to add this `guid` check and added it to the `update` functions.